src/Classes/ServiceAPI/MyRadio_Track.php

```diff
@@ -770,10 +770,21 @@ public function getSimilar() {
             }
             foreach ($data['similartracks']['track'] as $r) {
                 if ($r['match'] >= 0.25) {
+                    //Try to find an exact match
                     $c = self::findByOptions(['title' => $r['name'],
                                 'artist' => $r['artist']['name'],
                                 'limit' => 1,
-                                'digitised' => true]);
+                                'digitised' => true
+                                'precise' => true]);
+                    //Try to find a not-so-exact match
+                    if (!empty($c)) {
```

Comment:
if not empty is not the right condition to match. It probably will be empty if there are no matches, though that should be checked what findByOptions returns